### DISCUSSION TITLE
filter JSON parser - pattern not matched with data

### DISCUSSION THREAD

**Discussion by den-is** [fluent/fluentd#3950] [Q&A] [answered]
I'm trying to parse and extract keys from a JSON object
Using primitive Filter configuration.
fluentd 1.14.0 and all the latest plugins.

```ruby
<source>
  @type cloudwatch_logs
  @id star_dev_postgresql
  tag star_dev_postgresql
  fetch_interval 60
  log_group_name /aws/rds/cluster/star-dev/postgresql
  log_stream_name star-dev
  use_log_stream_name_prefix true
  include_metadata true
  add_log_group_name true
  log_group_name_key log_group
  use_log_group_name_prefix false
  format none
  <storage>
    @type local
  </storage>
  <web_identity_credentials>
    role_arn "#{ENV['AWS_ROLE_ARN']}"
    role_session_name fluentd-cw-star-dev-rds
    web_identity_token_file "#{ENV['AWS_WEB_IDENTITY_TOKEN_FILE']}"
  </web_identity_credentials>
</source>

<filter start_dev_postgresql>
  @type parser
  @id parse_cw_meta
  key_name metadata
  reserve_data true
  remove_key_name_field true
  # emit_invalid_record_to_error false
  <parse>
    @type json
  </parse>
</filter>
```

`metadata` is a key added by the CloudWatch source.

The error I'm getting:

> 2022-11-03 18:12:38 +0000 [warn]: #0 dump an error event: error_class=Fluent::Plugin::Parser::ParserError error="pattern not matched with data '{\"log_stream_name\"=>\"star-dev-2.1\", \"log_group_name\"=>\"/aws/rds/cluster/star-dev/postgresql\"}'" location=nil tag="star_dev_postgresql" time=2022-11-03 18:12:38.329077547 +0000 record={"message"=>"2022-11-03 18:12:02 UTC:192.168.150.180(55498):star_graphnode@graph_node:[3050]:LOG:  duration: 2958.044 ms  statement: vacuum (analyze) subgraphs.subgraph_deployment", "log_group"=>"/aws/rds/cluster/star-dev/postgresql", "metadata"=>{"log_stream_name"=>"star-dev-2.1", "log_group_name"=>"/aws/rds/cluster/star-dev/postgresql"}}

I'm not sure what kind of pattern is not matched?
`metadata` is a simple JSON object
```ruby
"metadata"=>{"log_stream_name"=>"star-dev-2.1", "log_group_name"=>"/aws/rds/cluster/star-dev/postgresql"}
```

Can't find answer in docs:
https://docs.fluentd.org/parser/json
https://docs.fluentd.org/configuration/parse-section

How do you extract fields from simple JSON objects without crazy regex/grok manipulations?

**Comment by fujimotos** [comment, marked as answer]
`metadata` is not JSON, but a Ruby object. You can tell that by
a pair of key and value is separated by `=>`, not `:`.

```ruby
"metadata"=>{"log_stream_name"=>"star-dev-2.1", ...
                             ====
```

So you have to parse nothing, because the message is already parsed!

**Reply by den-is** [reply]
OMG :/ ... indeed... actually i've noticed this ruby map structure, but I thought that it is just a representation of parsed JSON object in a record
